### DISCUSSION TITLE
Development vs Production

### DISCUSSION THREAD

**Discussion by varna** [supabase/supabase#4685] [Feature Requests]
Hello,

In our app (which doesn't use Supabase yet) we have two servers: **prod** and **dev**.
```js
export const BASE_URL =
  import.meta.env.PROD
    ? `https://our-app-name.com`
    : `https://our-app-name.dev`
```

During our development cycle:
1. Our backender is the first one to add some features to **dev** `https://api.our-app-name.dev`
2. Frontenders update their apps to work with new backend version, while using **dev** server `https://api.our-app-name.dev`
3. We release a **test** channel that uses `https://api.our-app-name.dev` to QA (iOS distributor, Android distributor and `https://our-app-name.dev` web app).
4. If green light is given, we copy **settings** from **dev** server to **prod** server and release updated version of apps and web app (`https://our-app-name.com`)

We use this approach to test out the possible synchronisation problems between different clients, that's why we need to use the same server. I also noticed that most of `auth` methods of supabase that I tested doesn't work in dockerized localhost. So I wanted to ask:

1. Can I host another Supabase instance to use as a dev server, or would that be against some rules?
2. Is there an easy way to copy settings from one Supabase instance to another?

**Comment by bohan0** [comment]
There is documentation on how to migrate the database data/schema ([Migrating between projects](https://supabase.com/docs/guides/database#migrating-between-projects)) but this does not copy the Storage objects and also does not copy the project settings (SMTP, Auth options, etc...)

**Reply by adamzerner** [reply]
It looks like the link for "Migrating between projects" is dead and I'm not seeing it on that webpage.

**Comment by adamzerner** [comment]
Bump on this. To elaborate on this and on [#1514](https://github.com/supabase/supabase/discussions/1514), I think it is a very common need to have different databases for development and production (and also often ones for different environments like QA and staging) but with the same schema, settings and configuration-related things.

@bohan0's [comment](https://github.com/supabase/supabase/discussions/4685#discussioncomment-1880377) above is the only thing I've found that addresses this need. However, having two (or more) separate projects where you have to manually keep the settings in sync is not ideal. It would be great if you could have one project but different databases for different environments (dev, QA, prod, etc.).
